haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -506,7 +507,29 @@ refsFromRenamed ctx declAlts (hsGroup, _, _, _) =
         TypeSig names _ _ ->
 #endif
             mapMaybe (\(L l n) -> give ctx (nameLocToRef n TypeDecl l)) names
+        (PatSynSig lnames _) ->
```

Comment:
See universeBi comment below, maybe you can get away with a catch-all clause + biPlate, like

   other -> (...map magic with nameLocToRef...) universeBi other